File: graphics_lab_final/DDA.py
import matplotlib.pyplot as plt
X1, Y1 = 2, 3
X2, Y2 = 12, 8
DX = X2 - X1
DY = Y2 - Y1
STEPS = max(abs(DX), abs(DY))
X_INC = DX / STEPS
Y_INC = DY / STEPS
X, Y = X1, Y1
plt.figure()
plt.grid(True)
plt.axis('equal')
plt.title('DDA Algorithm')
plt.xlabel('X-axis')
plt.ylabel('Y-axis')
X_VALS, Y_VALS = [], []
for I in range(STEPS + 1):
    X_VALS.append(round(X))
    Y_VALS.append(round(Y))
    X += X_INC
    Y += Y_INC
plt.plot(X_VALS, Y_VALS, 'ks', markersize=8, markerfacecolor='k')
plt.plot([X1, X2], [Y1, Y2], 'r--')
plt.show()

File: graphics_lab_final/DDA.m
clc; clear; close all;
X1 = 2; Y1 = 3;
X2 = 12; Y2 = 8;
DX = X2 - X1;
DY = Y2 - Y1;
STEPS = max(abs(DX), abs(DY));
X_INC = DX / STEPS;
Y_INC = DY / STEPS;
X = X1;
Y = Y1;
figure;
hold on;
grid on;
axis equal;
title('DDA Algorithm');
xlabel('X-axis'); ylabel('Y-axis');
for I = 1:STEPS+1
    plot(round(X), round(Y), 'ks', 'MarkerSize', 8, 'MarkerFaceColor', 'k');
    X = X + X_INC;
    Y = Y + Y_INC;
end
plot([X1 X2], [Y1 Y2], 'r--');
